src/show.jl

```diff
@@ -21,3 +21,42 @@ Base.show(io::IO, A::DiskGeoArray) = begin
     end
     print(io, "\n  From file: $(filename(A))")
 end
+
+function Base.show(io::IO, stack::AbstractGeoStack)
+    printstyled(io, "$(Base.typename(typeof(stack)))", color=:blue)
+    print(io, " with $(length(keys(stack))) field(s)")
+    applicable(filename, stack) && print(io, ": $(filename(stack))")
+    print(io, '\n')
+
+    for var in keys(stack)
+        printstyled(io, " $var", color=:green)
+
+        field_dims = dims(stack, var)
+        if length(field_dims) > 0
+            print(io, " with dimension(s) ")
+            for (d, dim) in enumerate(field_dims)
+                printstyled(io, "$(GeoData.name(dim))($(length(dim)))", color=:red)
+                d != length(field_dims) && print(io, ", ")
+            end
+        end
+        print(io, '\n')
+    end
+
+    n_windows = length(stack.window)
+    if n_windows > 0
+        print(io, "and with $n_windows window(s):\n")
```

Comment:
This can just be "with window:" - the whole tuple is the window, so its not pluralised

Reply:
Done.